scripts/generate-releases-manifest.mjs

```diff
@@ -121,9 +134,11 @@ if ( isDevBuild ) {
 	};
 
 	// Windows
+	const windowsReleaseInfo = await getWindowsReleaseInfo();
 	releasesData[ 'dev' ][ 'win32' ] = {
-		sha: currentCommit,
-		url: `${ cdnURL }/${ baseName }-win32-${ currentCommit }.exe`,
+		sha: windowsReleaseInfo.sha1,
+		url: `${ cdnURL }/${ baseName }-win32-v${ version }-${ currentCommit }-full.nupkg`,
```

Comment:
> Would it be enough to keep doing our versioning as `x.y.z-beta.N` internally (for GitHub Releases and tags for example, as well as for the value in `package.json` probably?) but _**only**_ do a simple replacement of `beta.` with `beta-` in the `version` variable _when interpolating it_ into the URL?

Good idea. I was also thinking of using `-` as a replacement for `.` in the version value.

> Or does Squirrel _require_ somehow that those two (the version it extracts via `ToSemanticVersion` and the one in `package.json` and our GitHub Releases / git tags etc) match, and thus doing that replacement only in the URL's filename won't be enough?

As far as I know, Squirrel extracts the version from the filename (which is included in the URL), so if we ensure that we use `-` instead of `.` there, it should be fine.

Reply:
Well what I'm wondering is, if Squirrel compares the new version (extracted from the filename) with the currently installed version on the user's machine (which will probably come from the value from `package.json`), that might not work fully as expected, as it will try to semver-compare a current version of `1.2.3-beta.4` (from `package.json` of installed app) with `1.2.3-beta-3` (extracted from file name), right? Not sure how semver-comparison behaves in such mixed-format versions and which one it considers to be the most recent then…? 🤔

Reply:
We could add a dot to regex to fix it in Squirrel, similar to https://github.com/Squirrel/Squirrel.Windows/commit/c4e875bdfc888ca0080c990e1090a966e8591f5f, but I guess it can take time to get this merged.

To simplify things, do you see anything against dropping the dot and switching to the `1.0.3-beta1` format? According to SemVer, it seems valid and is correctly interpreted by PHP's `version_compare()`.

It would be great to keep beta releases working as they work now and not only use those internally. Now we (or users) can install beta version and benefit from auto-update process when a) new beta is released b) final version is released.

Reply:
Yep would work too.

My suggestion above was more because I'm not aware of how you are generating a new beta in this repo.
Are you updating the version in `package.json` **manually** then doing a new tag **manually** to trigger a build, or is the version bump and tag triggered by some automation / make target / Fastlane lane?

In out other app products we have automation for three new beta, which is in charge for the version bump … and expects this `x.y-beta.N` format for beta; that's why I assumed this was the case with studio too, in which case we'd have to update our tooling around the version bump too to match whatever new version format we'd want to use. While if all of the version bumping is manual and there's no consequence (no other tooling risking to break) then it's just a simple team decision on the convention when they change the version in `package.json` manually 🙂

Reply:
> Are you updating the version in package.json manually then doing a new tag manually to trigger a build, or is the version bump and tag triggered by some automation / make target / Fastlane lane?

It's manual now (see https://github.com/Automattic/studio/blob/trunk/docs/release-process.md), so we have freedom here. We can write a simple script at some point to automate that, though. There is no tooling that would expect `beta.1` vs `beta1`.

Reply:
Perfect 👌
Let's choose a version name convention that works with Squirrel out of the box then 👍

Reply:
💡 Would it be worth adding a small piece of code in the scripts to validate that the version set in `package.json` uses a format recognized by the Squirrel regex?

So that someone who didn't see our remember this conversation in the future doesn't risk using something like `x.y-beta.N` out of habit and breaks the update logic accidentally?